### PR TITLE
Optimize Panorama view layout

### PR DESCRIPTION
**Pull request by thomasm0**

### CODE REVIEW

File: src/app/pages/MapPage/MapPage.tsx [hunk 1]
```diff
@@ -171,6 +194,12 @@ const MapPage: FunctionComponent = () => {
     mapInstanceRef.current?.setZoom(zoom)
   }, [zoom])
 
+  useMemo(() => {
```
Comment (thomasm0) [inline src/app/pages/MapPage/MapPage.tsx:197]: It seems like this should be a useEffect but whenever I try with useEffect you start getting issues with the mini-map positioning being wrong. I looked through the useEffect order to see if the map is getting repositioned by something else but cannot see anything.

Comment (chrisvanmook) [reply src/app/pages/MapPage/MapPage.tsx:197]: let's look into this because this should be in a useeffect indeed

Comment (thomasm0) [reply src/app/pages/MapPage/MapPage.tsx:197]: For reference - `flyTo` can be used to panTo a location and zoom (first argument latlng, second arg is zoom level)

For example `mapInstanceRef.current.flyTo(position, 12)`

Comment (thomasm0) [reply src/app/pages/MapPage/MapPage.tsx:197]: The following changes to the MapPage.tsx useEffects seems to work, however, @chrisvanmook maybe you know some more specific map links where the zoom level is predefined. I haven't looked at the param handling with this either:

```
useEffect(() => {
  if (mapInstanceRef.current) {
    mapInstanceRef.current.invalidateSize()
  }
}, [panoActive, panoFullScreen, panoPitch, mapInstanceRef])

useEffect(() => {
  if (mapInstanceRef.current && position) {
    if (panoFullScreen) {
      mapInstanceRef.current.flyTo(position, 12)
    } else {
      mapInstanceRef.current.panTo(position)
    }
  }
}, [panoActive, panoFullScreen])

useEffect(() => {
  // mapInstanceRef.current?.setZoom(zoom)
}, [zoom])

useEffect(() => {
  if (mapInstanceRef && panoFullScreen) {
    setZoom(12)
  }
}, [panoFullScreen])
```

**Review by chrisvanmook — APPROVED** [review]

**Pull request merged**